### PR TITLE
feat: show an approximation of transaction time

### LINKED ISSUE #1428: The speed of xDAI is truly amazing.  Why not clearly show transaction times in Blockscout
One of the most amazing value propositions for xDAI is the incredible speed of transactions.  ⚡️ Why not clearly highlight this in Blockscout on each transaction. Imagine being able to look up a transaction and see something like:

"**Transaction time: took 1.1045 seconds to complete**"

I believe that would really encourage more usage, and would set a very high bar for other chains and coins to follow.

### PR DESCRIPTION
**Pull request by zachdaniel**
Resolves #1428

## Motivation

We want to show information on the time it took a transaction to complete.

We can't get this information completely accurately. All we can do is record the last time we checked for pending transactions, and the current time when we are fetching pending transactions. Using that information, we can determine a range in which the transaction must have started, and show that range in the explorer. If the process monitoring the pending transaction dies or if the application restarts, we will not be able to determine that information. We also won't be able to determine that information for any transaction that has already happened.

We should decide if we want to show anything at all on the pages where we have no information on how long the transaction took to complete. Should we show nothing? The current implementation shows `Unknown`. I'm also currently looking at ways to test this interaction in the pending transaction fetcher. All in all it came out to be a *much* simpler change than I was expecting.

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/views/transaction_view.ex [hunk 1]
```diff
@@ -29,6 +30,52 @@ defmodule BlockScoutWeb.TransactionView do
 
   def value_transfer?(_), do: false
 
+  def processing_time_duration(%Transaction{block: nil}) do
+    :pending
+  end
+
+  def processing_time_duration(%Transaction{earliest_processing_start: nil}) do
+    {:ok, "Unknown"}
```
Comment (acravenho) [inline apps/block_scout_web/lib/block_scout_web/views/transaction_view.ex]: If the time is `Unknown`, I would prefer that we didn't show the tx time

**Comment by zachdaniel** [comment]
Adding some more tests to this now.

**Comment by acravenho** [comment]
@zachdaniel Everything looks great! Good work. I received this error that isn't related to this PR but it would be good to fix asap:

```
2019-02-19T21:17:03.891 [error] #PID<0.29307.0> running BlockScoutWeb.Endpoint (connection #PID<0.29194.0>, stream id 4) terminated
Server: localhost:4000 (http)
Request: GET /address/0x6109dc5ba1159ea6974c4b15ea9e68a180f85753/validations
** (exit) an exception was raised:
    ** (ArgumentError) assign @coin_balance_status not available in eex template.

Please make sure all proper assigns have been set. If this
is a child template, ensure assigns are given explicitly by
the parent template as they are not automatically forwarded.

Available assigns: [:address, :conn, :current_path, :exchange_rate, :transaction_count, :validation_count, :view_module, :view_template]

        (phoenix_html) lib/phoenix_html/engine.ex:132: Phoenix.HTML.Engine.fetch_assign!/2
        (block_scout_web) lib/block_scout_web/templates/address/overview.html.eex:77: BlockScoutWeb.AddressView."overview.html"/1
        (block_scout_web) lib/block_scout_web/templates/address_validation/index.html.eex:2: BlockScoutWeb.AddressValidationView."index.html"/1
        (block_scout_web) lib/block_scout_web/templates/layout/app.html.eex:39: BlockScoutWeb.LayoutView."app.html"/1
        (phoenix) lib/phoenix/view.ex:399: Phoenix.View.render_to_iodata/3
```

**Comment by zachdaniel** [comment]
@acravenho resolved the issue you mentioned in #1462

**Pull request merged**